README.rst

```diff
@@ -580,6 +631,88 @@ Example pipeline yaml using a pipe:
 See a worked example `for shell power here
 <https://github.com/pypyr/pypyr-example/tree/master/pipelines/shell.yaml>`__.
 
+pypyr.steps.tar
+^^^^^^^^^^^^^^^
+Archive and/or extract tars with or without compression.
+
+At least one of these context keys must exist:
+
+- tarExtract
+- tarArchive
+
+Optionally, you can also specify the tar compression format with
+``context['tarFormat']``. If not specified, defaults to *lzma/xz*
+Available options:
+
+- '' - no compression
+- gz (gzip)
+- bz2 (bzip2)
+- xz (lzma)
+
+This step will run whatever combination of Extract and Archive you specify.
+Regardless of combination, execution order is Extract, Archive.
+
+Never extract archives from untrusted sources without prior inspection. It is
+possible that files are created outside of path, e.g. members that have
+absolute filenames starting with "/" or filenames with two dots "..".
+
+See a worked example `for tar here
+<https://github.com/pypyr/pypyr-example/tree/master/pipelines/tar.yaml>`__.
+
+tarExtract
+""""""""""
+``context['tarExtract']`` must exist. It's a dictionary.
+
+keys are the path to the tar to extract.
+
+values are the destination paths.
+
+You can use {key} substitutions to format the string from context.
+
+.. code-block:: yaml
+
+  key1: here
+  key2: tar.xz
+  tarExtract:
+    - in: path/to/my.tar.xz
+      out: /path/extract/{key1}
+    - in: another/{key2}
+      out: .
+
+This will:
+
+- Extract *path/to/my.tar.xz* to */path/extract/here*
+- Extract *another/tar.xz* to the current execution directory
+
+  - This is the directory you're running pypyr from, not the pypyr pipeline
+    working directory you set with the ``--dir`` flag.
+
+tarArchive
+""""""""""
+``context['tarArchive']`` must exist. It's a dictionary.
+
+keys are the paths to archive.
+
+values are the destination output paths.
+
+You can use {key} substitutions to format the string from context.
+
+.. code-block:: yaml
+
+  key1: destination.tar.xz
+  key2: value2
+  tarArchive:
+    - in: path/{key2}/dir
+      out: path/to/{key1}
+    - in: another/my.file
+      out: ./my.tar.xz
+
+This will:
+
+- Archive directory *path/value2/dir* to *path/to/destination.tar.xz*,
+- Archive file *another/my.file* to *./my.tar.xz*
+
+
 Roll your own step
 ------------------
```

Comment:
Yes. I'm still brooding a bit on what the structure should be, but yes, I'd say it's say there's nothing "a bit" about the bestial nature of it! Prob only keep the "run your 1st pipeline" in README, and shift the rest off into gh pages. I prob want to talk to you about your experience w, um, I just forgot the name, vs the RSTs that python prefers.